### ISSUE TITLE
Payara EJB Timer Service deployment failure

### ISSUE THREAD

**Issue by ctabin** [payara/Payara#6148]
### Brief Summary

Hello,

Since this night, we are unable to start our payara-embedded-all application because it crashes when our EAR is deployed with the following exception:

```
java.io.IOException: org.xml.sax.SAXParseExceptionpublicId: http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/persistence/persistence_2_1.xsd; lineNumber: 1; columnNumber: 57; Deployment descriptor file META-INF/persistence.xml in archive [classes].  s4s-elt-character: Non-whitespace characters are not allowed in schema elements other than 'xs:appinfo' and 'xs:documentation'. Saw 'Oracle Access Manager Operation Error'.
        at org.glassfish.javaee.core.deployment.DolProvider.processDOL(DolProvider.java:229)
        at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:246)
        at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:97)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.loadDeployer(ApplicationLifecycle.java:1161)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.loadDeployers(ApplicationLifecycle.java:1129)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainerInfos(ApplicationLifecycle.java:944)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.prepare(ApplicationLifecycle.java:453)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:658)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:652)
        at org.glassfish.ejb.persistent.timer.PersistentEJBTimerService.deployEJBTimerService(PersistentEJBTimerService.java:1463)
        at org.glassfish.ejb.persistent.timer.PersistentEJBTimerService.initEJBTimerService(PersistentEJBTimerService.java:1369)
        at org.glassfish.ejb.persistent.timer.DatabaseEJBTimerService.initPersistentTimerService(DatabaseEJBTimerService.java:98)
        at com.sun.ejb.containers.EJBTimerService.initPersistentTimerService(EJBTimerService.java:423)
        at com.sun.ejb.containers.EJBTimerService.getEJBTimerService(EJBTimerService.java:257)
        at com.sun.ejb.containers.BaseContainer.initialize(BaseContainer.java:871)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.glassfish.ejb.startup.EjbApplication.initialize(EjbApplication.java:250)
        at org.glassfish.internal.data.EngineRef.initialize(EngineRef.java:203)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at org.glassfish.internal.data.ApplicationInfo.lambda$initialize$0(ApplicationInfo.java:393)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.glassfish.internal.data.ApplicationInfo.initialize(ApplicationInfo.java:393)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.initialize(ApplicationLifecycle.java:626)
        at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:616)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:556)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:552)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.base/javax.security.auth.Subject.doAs(Subject.java:376)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:551)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:582)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:574)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.base/javax.security.auth.Subject.doAs(Subject.java:376)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:573)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1497)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1869)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1755)
        at com.sun.enterprise.admin.cli.embeddable.DeployerImpl.deploy(DeployerImpl.java:131)
        ...
Caused by: org.xml.sax.SAXParseExceptionpublicId: http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/persistence/persistence_2_1.xsd; lineNumber: 1; columnNumber: 57; Deployment descriptor file META-INF/persistence.xml in archive [classes].  s4s-elt-character: Non-whitespace characters are not allowed in schema elements other than 'xs:appinfo' and 'xs:documentation'. Saw 'Oracle Access Manager Operation Error'.
        at com.sun.enterprise.deployment.io.DeploymentDescriptorFile.read(DeploymentDescriptorFile.java:322)
        at com.sun.enterprise.deployment.archivist.ExtensionsArchivist.open(ExtensionsArchivist.java:178)
        at com.sun.enterprise.deployment.archivist.PersistenceArchivist.readPersistenceDeploymentDescriptor(PersistenceArchivist.java:157)
        at com.sun.enterprise.deployment.archivist.WarPersistenceArchivist.open(WarPersistenceArchivist.java:99)
        at com.sun.enterprise.deployment.archivist.Archivist.readRestDeploymentDescriptors(Archivist.java:414)
        at com.sun.enterprise.deployment.archivist.Archivist.readDeploymentDescriptors(Archivist.java:395)
        at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:271)
        at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:280)
        at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:240)
        at com.sun.enterprise.deployment.archivist.ApplicationFactory.openArchive(ApplicationFactory.java:161)
        at org.glassfish.javaee.core.deployment.DolProvider.processDOL(DolProvider.java:221)
        ... 93 more
Caused by: org.xml.sax.SAXParseException; systemId: http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/persistence/persistence_2_1.xsd; lineNumber: 1; columnNumber: 57; s4s-elt-character: Non-whitespace characters are not allowed in schema elements other than 'xs:appinfo' and 'xs:documentation'. Saw 'Oracle Access Manager Operation Error'.
        at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
        at org.apache.xerces.util.ErrorHandlerWrapper.error(Unknown Source)
        at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
        at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
        at org.apache.xerces.impl.xs.opti.SchemaDOMParser.characters(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanContent(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at org.apache.xerces.impl.xs.opti.SchemaParsingConfig.parse(Unknown Source)
        at org.apache.xerces.impl.xs.opti.SchemaParsingConfig.parse(Unknown Source)
        at org.apache.xerces.impl.xs.opti.SchemaDOMParser.parse(Unknown Source)
        at org.apache.xerces.impl.xs.traversers.XSDHandler.getSchemaDocument(Unknown Source)
        at org.apache.xerces.impl.xs.traversers.XSDHandler.parseSchema(Unknown Source)
        at org.apache.xerces.impl.xs.XMLSchemaLoader.loadSchema(Unknown Source)
        at org.apache.xerces.impl.xs.XMLSchemaValidator.findSchemaGrammar(Unknown Source)
        at org.apache.xerces.impl.xs.XMLSchemaValidator.handleStartElement(Unknown Source)
        at org.apache.xerces.impl.xs.XMLSchemaValidator.startElement(Unknown Source)
        at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanStartElement(Unknown Source)
        at org.apache.xerces.impl.XMLNSDocumentScannerImpl$NSContentDispatcher.scanRootElementHook(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
        at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
        at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
        at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at org.apache.xerces.jaxp.SAXParserImpl.parse(Unknown Source)
        at com.sun.enterprise.deployment.io.DeploymentDescriptorFile.read(DeploymentDescriptorFile.java:312)
```

It seems that it is because of [this file](https://github.com/payara/Payara/blob/Payara6/appserver/ejb/ejb-timer-service-app/src/main/webapp/WEB-INF/classes/META-INF/persistence.xml) that depends externally on Oracle which gives an error.

Would it be possible to integrate the XSD directly in the JAR or move to jakarta ?

Is there any workaround until this issue is fixed (or Oracle site come up) ?

Thanks.

### Expected Outcome

There should be no external dependency on third-party websites that causes an unsolvable failure during startup.

### Current Outcome

Unable to deploy any application with EJB timer (unless the Oracle website is fixed).

Currently, the [URL](http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/persistence/persistence_2_1.xsd) gives:
```
Oracle Access Manager Operation Error

The WebGate plug-in is unable to contact any Access Servers.

Contact your website administrator to remedy this problem.
```

### Reproducer

Any application that uses EJB timer will fail to deploy.

### Operating System

Linux & Windows

### JDK Version

JDK 17

### Payara Distribution

Payara Embedded All 6.2022.2

**Comment by shub8968** [comment]
Hi @ctabin,

Kindly refer to this blog - https://blog.payara.fish/troubleshooting-deploying-to-payara

**Comment by ctabin** [comment]
Hi @shub8968,

Thanks for your message. That's indeed a (temporary) solution.
Still, I suggest this dependency should be removed as per #6149.

**Comment by ctabin** [comment]
Should be solved by #6149.
